### PR TITLE
Closes #25 - Added MySQL prepost strategy & documentation

### LINKED ISSUE #25: Add an example for pre and post scripts
An example illustrating how to create a prepost strategy would help new users to create his own strategies

### PR DESCRIPTION
**Pull request by SrZorro**

### CODE REVIEW

File: prepost_strategies/mysql/prexecute/mysqldump.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+source /prexecute/utils/check-env.sh
+
+if [ "$BACKUP_TYPE" = "backup" ] || [ "$BACKUP_TYPE" = "backupIncremental" ] || [ "$BACKUP_TYPE" = "backupFull" ]; then
```
Comment (blacklabelops) [inline prepost_strategies/mysql/prexecute/mysqldump.sh]: You should use a case directive here: http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_07_03.html

File: prepost_strategies/mysql/prexecute/utils/check-env.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+## 
+ # @private
+ # Executed if a variable is not set or is empty string
+ # @param1 - Name of the failed environment variable
+ ##
+function _check_env_failed {
+    echo "[!] ENV VAR $1 IS NOT SET"
+    printf "\n====ENVIRONMENT VARIABLES FAILED====\n\n"
```
Comment (blacklabelops) [inline prepost_strategies/mysql/prexecute/utils/check-env.sh]: This is called `flagging - Clean Code`. Do not use upper case chars and symbolic characters in order to make it appear `better` on logs. Does not work when everybody documents like that.

**Comment by blacklabelops** [comment]
Thanks for the example. Please take a note of my comments.

**Comment by SrZorro** [comment]
In my use case I also need a MongoDB strategy, tomorrow when i do it i can also push it as another example and i can fix the problems pointed in your comments. 

For the second comment how would you do it? Without all the fancy symbolic characters and in lowercase?

**Pull request merged**